--- nvidia-settings-1.0/Makefile~	2006-12-25 20:48:25.000000000 +0200
+++ nvidia-settings-1.0/Makefile	2006-12-26 21:46:56.000000000 +0200
@@ -118,7 +118,7 @@
 GTK_CFLAGS := $(shell $(PKG_CONFIG) --cflags gtk+-2.0)
 GTK_LDFLAGS := $(shell $(PKG_CONFIG) --libs gtk+-2.0)
 
-X11_LIBS := $(X11_LIB_DIRS) -Wl,-Bstatic -lXxf86vm -Wl,-Bdynamic -lX11 -lXext
+X11_LIBS := $(X11_LIB_DIRS) -lXxf86vm -lX11 -lXext
 
 XNVCTRL_LIB := src/libXNVCtrl/libXNVCtrl.a
 
