--- nvidia-settings-1.0/Makefile~	2005-12-15 03:57:49 +0300
+++ nvidia-settings-1.0/Makefile	2005-12-26 18:57:23 +0300
@@ -118,7 +118,7 @@ X11R6_CFLAGS = -I $(X11R6_INC_DIR)
 GTK_CFLAGS := $(shell $(PKG_CONFIG) --cflags gtk+-2.0)
 GTK_LDFLAGS := $(shell $(PKG_CONFIG) --libs gtk+-2.0)
 
-X11R6_LIBS := -L $(X11R6_LIB_DIR) -Wl,-Bstatic -lXxf86vm -Wl,-Bdynamic -lX11 -lXext
+X11R6_LIBS := -L $(X11R6_LIB_DIR) -lXxf86vm -lX11 -lXext
 
 XNVCTRL_LIB := src/libXNVCtrl/libXNVCtrl.a
 
