--- nvidia-settings-515.48.07/src/Makefile.orig	2022-05-27 06:00:06.000000000 +0200
+++ nvidia-settings-515.48.07/src/Makefile	2022-06-01 21:50:14.917004206 +0200
@@ -42,10 +42,10 @@
 endif
 ifeq (1,$(WAYLAND_AVAILABLE))
   ifndef WAYLAND_CFLAGS
-    WAYLAND_CFLAGS := $(shell $(PKG_CONFIG) --cflags )
+    WAYLAND_CFLAGS := $(shell $(PKG_CONFIG) --cflags wayland-client)
   endif
   ifndef WAYLAND_LDFLAGS
-    WAYLAND_LDFLAGS := $(shell $(PKG_CONFIG) --libs )
+    WAYLAND_LDFLAGS := $(shell $(PKG_CONFIG) --libs wayland-client)
   endif
   BUILD_WAYLANDLIB = 1
 endif
